### ISSUE TITLE
uBlock Origin stopped working on Youtube, Spotify

### ISSUE THREAD

**Issue by shytikov** [gorhill/uBlock#675]
This is invitation to discussion, I'm still working on identifying reason of the issue. But here is the story.

It seems to be if URL is requested through `PepperFlash` plugin in Chrome, they bypass `uBlock0` filters.

I can see in developers tools, that requests sent from Flash succeeds, while the same URLs called from browser's address bar shows blocking notification.

It happened when I have updated Chrome to version 45. They promised to abandon third-party plugins and they did this (once again). Yesterday when I was running website through Adobe Flash, addresses was blocked, but today PepperFlash bypasses all blocking easily.

Have someone else noticed the same behavior?

**Comment by gorhill** [comment]
> if URL

Any test case (specific URL) so that anyboby can try and reproduce?

**Comment by shytikov** [comment]
Here is screen. It's Spotify Web player.

![Das Model](https://cloud.githubusercontent.com/assets/1185905/9631779/83b87f10-5183-11e5-9930-269098fd42c3.png)

It's sends some tracking (successfully!!! note status `200`!!!) to [following URL](https://stats.g.doubleclick.net/__utm.gif?utmwv=5.6.6dc&utms=4&utmn=1778985440&utmhn=play.spotify.com&utmt=event&utme=5(upsell*show_version*mft)(1)&utmcs=UTF-8&utmsr=1920x1080&utmvp=1920x989&utmsc=24-bit&utmul=en-us&utmje=1&utmfl=18.0%20r0&utmdt=%E2%96%B6%20Das%20Model%20-%20The%20Space%20Rangers%20-%20Spotify&utmhid=1548972005&utmr=-&utmp=%2Fartist%2F4YVK4q8FoSDmaV6ikZaKPZ&utmht=1441198597910&utmac=UA-5784146-11&utmcc=__utma%3D200548975.277170573.1441176423.1441176423.1441198552.3%3B%2B__utmz%3D200548975.1441176423.1.1.utmcsr%3D(direct)%7Cutmccn%3D(direct)%7Cutmcmd%3D(none)%3B&utmjid=&utmu=6ACAAAAAAAAAAAAAAAAAAAAE~).

The URL will be immediately blocked by `uBlock0` as soon as you click on it.

**Comment by gorhill** [comment]
What is the URL of that web page? I want an exact URL, so that I do not have to second guess whether I have a good case or not: I want the same test case as yours.

**Comment by gitarra** [comment]
Related? https://github.com/gorhill/uBlock/issues/633

**Comment by gorhill** [comment]
@shytikov Also, could you look into what uBlock's logger is reporting? I am wondering whether the network request is reported as behind-the-scene network request.

**Comment by gorhill** [comment]
Ok I am guessing the URL is <https://play.spotify.com/artist/4YVK4q8FoSDmaV6ikZaKPZ> (Space Rangers).

**Comment by shytikov** [comment]
@gorhill Sorry for delay, yes, the URL is correct. 

@gitarra 100% related.

**Comment by gorhill** [comment]
With Google Chrome 46.0.2490.6 dev (64-bit) on Linux, I can't reproduce here. I do see a request to `stats.g.doubleclick.net`, but it is properly blocked:

![a](https://cloud.githubusercontent.com/assets/585534/9632480/8359ea64-5155-11e5-8ac7-026958b63ae3.png)

**Comment by shytikov** [comment]
@gorhill can you give me some hints how I can get information from `uBlock0` log?

**Comment by gorhill** [comment]
I see differences between my result with your result:

- Your request to `stats.g.doubleclick.net` is for a GIF, mine was for a script.
- The "initiator" of your request was "other", mine was `spotify.web.client.js:309`.

What are your uBlock settings? Please provide _all_ settings which differ from default ones (I use default settings in my attempt to reproduce).

**Comment by shytikov** [comment]
Default settings. Reseted them once again right not to make sure they are stock. 

Can you please check on Linux, which version of flash are you using? `Adobe` or `Pepper`?

**Comment by gorhill** [comment]
> which version of flash are you using?

Pepper: Adobe Flash Player - Version: 19.0.0.142

**Comment by shytikov** [comment]
In my case it's Pepper: Adobe Flash Player - Version: 18.0.0.232

**Comment by shytikov** [comment]
I believe on Windows Pepper runs in container so separate, so `uBlock0` cannot access it.

**Comment by gorhill** [comment]
> so `uBlock0` cannot access it

It does not matter, the browser is supposed to pass network requests to uBlock through its webRequest API, it does not matter where the network request comes from.

**Comment by shytikov** [comment]
Is there any test cases I can run on my PC to determine the reason of the issue?

**Comment by gorhill** [comment]
I think you should look in uBlock's logger first to see if the offending network request is seen or not by uBlock. If it is a behind-the-scene request, it won't be blocked by default, and this could be what you are experiencing.

**Comment by shytikov** [comment]
`uBlock0` is empty. You can see even on my screenshot number of blocked elements is empty. I have checked log too. It's empty. Not a record.It seems all requests are bypassed.

Before that Spotify triggered a hundreds of blocked items.

**Comment by gorhill** [comment]
> Not a record there. It seems all requests are passing by.

Ok, so it does seem you are experiencing exactly the same problem as reported here: <https://code.google.com/p/chromium/issues/detail?id=526413>. Someone posted a screenshot of the logger, and this suggests that for whatever reasons, [chrome.webRequest API](https://developer.chrome.com/extensions/webRequest) stops working.

**Comment by gorhill** [comment]
@shytikov Do you have uBlock Origin's _"Collect errors"_  checked in `about:extensions`? If not, could you check it and once it's checked, are there any errors reported?

**Comment by gorhill** [comment]
Related: #594.

**Comment by shytikov** [comment]
@gorhill related :100: 

I will try to get you more information with *"Collect errors"* enabled...

**Comment by gorhill** [comment]
Side note, there has been some work done in `chrome.webRequest`-related code in the last four weeks: <https://chromium.googlesource.com/chromium/src/+log/master/extensions/browser/api/web_request/web_request_api.cc>.

**Comment by harshanvn** [comment]
this is very bad news, if chrome disables ability to view net requests made by other extensions...:(

Hope this will not be true with Firefox, with their web extensions api..

**Comment by gorhill** [comment]
Do you have _"Continue running background apps when Chromium is closed"_ checked in the browser's advanced settings?

**Comment by tvanderka** [comment]
This happens if you have youtube app in chrome, remove it and uBlock starts working.

The new webrequest filtering code treats those apps like extensions and filters the requests out.

**Comment by gorhill** [comment]
@tvanderka Thanks for the tip! I will share with others.

**Comment by gorhill** [comment]
@tvanderka I launched Chrome 47 and went to install the [Youtube app](https://chrome.google.com/webstore/detail/youtube/blpcfgokakmgnkcojhhkbfbldkacnbeo/) to see if I could now reproduce the issue, but apparently it is already installed, and yet I am still unable to see cideo ads.

**Comment by Luxocracy** [comment]
Just posting to confirm that the solution that @tvanderka solved the issues I've been having with youtube ads as of late.

**Comment by gorhill** [comment]
Confirmed working by other users: <https://forums.lanik.us/viewtopic.php?f=62&t=25033&start=15>.

**Comment by gorhill** [comment]
So to recapitulate:

- [Regression bug on Chromium 45 and higher](https://code.google.com/p/chromium/issues/detail?id=526413#c6) (affects all blockers).
    - Roughly, side effect of a fix related to a security issue (which is not yet public).
- Workaround as posted by @tvanderka:
    - Open a new tab.
    - Type `chrome://apps` in the address bar.
    - Remove Youtube app.

**Comment by Rob--W** [comment]
@shytikov Did you install the [Spotify app](https://chrome.google.com/webstore/detail/spotify-music-for-every-m/cnkjkdjlofllcpbemipjbcpfnglbgieh)? Due to a bug in Chromium (https://code.google.com/p/chromium/issues/detail?id=526413), extensions cannot block requests originating from websites for which you have installed a "hosted app".

The work-around for now is to disable (or remove) the Spotify app.

**Comment by shytikov** [comment]
@Rob--W yes, I have Spotify app... After removing it started working! Thanks!

**Comment by JulianNorton** [comment]
I'm still gettings ads from youtube even though I've removed the youtube app.
![screen shot 2015-09-09 at 7 56 56 pm](https://cloud.githubusercontent.com/assets/2412992/9777031/027a97b2-572d-11e5-80d0-cdc2489da586.png)
![Uploading Screen Shot 2015-09-09 at 7.57.00 PM.png…]()

**Comment by gorhill** [comment]
[Logger](https://github.com/gorhill/uBlock/wiki/The-logger).

**Comment by j3mdamas** [comment]
*this is just a thank you note*
Thanks a lot about this tip on removing the Youtube and Spotify apps! I am not a user of uBlock origin, but this thread helped me solving the problem with the tracker blocker I use.

**Comment by Rob--W** [comment]
FYI, at the next release of Chrome 45 and 46, the problem will be fixed. All patches have landed in the right branches, the only remaining to-do is actually publishing a new version of Chrome.

**Comment by gorhill** [comment]
Duplicate of #633.

**Comment by pawel2105** [comment]
I'd like to chime in that I'm seeing ads on YouTube using 1.1.1 on FF 40.0.3.

**Comment by gorhill** [comment]
If it happens again, look in the [logger](https://github.com/gorhill/uBlock/wiki/The-logger) to see what is happening. Also a look at the browser console (<kbd>Ctrl</kbd>-<kbd>Shift</kbd>-<kbd>j</kbd>) to see if there are errors reported.
